packages/roosterjs-content-model-api/lib/modelApi/common/queryContentModel.ts

```diff
@@ -0,0 +1,116 @@
+import type {
+    ContentModelBlockType,
+    ContentModelSegmentType,
+    ReadonlyContentModelBlock,
+    ReadonlyContentModelBlockGroup,
+    ReadonlyContentModelParagraph,
+    ReadonlyContentModelSegment,
+    ReadonlyContentModelTable,
+} from 'roosterjs-content-model-types';
+
+/**
+ * Options for queryContentModel
+ */
+export interface QueryContentModelOptions<T> {
+    /**
+     * The type of block to query @default 'Paragraph'
+     */
+    type?: ContentModelBlockType;
+
+    /**
+     * The type of segment to query
+     */
+    segmentType?: ContentModelSegmentType;
+
+    /**
+     * Optional selector to filter the blocks/segments
+     */
+    selector?: (element: T) => boolean;
```

Comment:
`(element: ContentModelType) => element is Target` this is the type of the selector we used in our implementation, I'm not sure if it's a good one though. Just an idea.